scrubadub/detectors/date_of_birth.py

```diff
@@ -22,22 +22,29 @@ class DoBDetector(Detector):
     name = 'dob'
 
     def __init__(self, context_before: Optional[int] = None, context_after: Optional[int] = None,
-                 min_age_years: Optional[int] = None,
+                 min_age_years: Optional[int] = None, filter_type: Optional[str] = None,
                  **kwargs):
-        super(DoBDetector, self).__init__(**kwargs)
+
         if context_before is None:
             self.context_before = int(2)
         if context_after is None:
             self.context_after = int(1)
         if min_age_years is None:
             self.min_age_years = int(18)
+        if filter_type is None:
+            self.filter_type = 'context_dob'
 
         self.trigger_words = ['birth', 'born', 'dob', 'd.o.b.']
+        super(DoBDetector, self).__init__(**kwargs)
 
-    def iter_filth(self, text: str, document_name: Optional[str] = None):
-        # splitting lines using linebreaks, which is then used for counting
-        lines = text.split('\n')
-
+    def context_dob(self, text: str, document_name: Optional[str] = None):
+        """ Contextual Date of Birth filter -
+        only look for dates that are in close proximity with the words similar to birth
+        """
+        if isinstance(text, list):
```

Comment:
Text is defined as a string, it can't be a list.

Reply:
as discussed - errors are being raised in the right place, let's revisit later